### ISSUE TITLE
Bug: [no-namespace] is used with [prefer-namespace-keyword] within `recommended`

### ISSUE THREAD

**Issue by Lonli-Lokli** [typescript-eslint/typescript-eslint#5066]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=4.7.1-rc&sourceType=module&code=KYDwDg9gTgLgBAWwgEwK4BthwMoGMAWwCAhgM5wDeAsAFBxyiSxy4QB2p8AwhAmMWwCePNjCjFc8ALxwA5DGCdZAbloBfIA&rules=N4IgAgLgngDgpgZwMYCcCWMIFpEBs0B2EA9DCnAGZwpYECGAtojHUnFgNZxQDuA9igAmIAFwhqKASAA04aPGTpMOBPiLECfWo2as4o8SkkoQAXyA&tsConfig=N4XyA

### Repro Code

```typescript
export module Schemas {
  export const CompanyContract = 'test';
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/prefer-namespace-keyword": "error",
    "@typescript-eslint/no-namespace": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Should work one of them, now both rules are included in recommmended

### Actual Result

namespace Schemas
Use 'namespace' instead of 'module' to declare custom TypeScript modules.ESLint(@typescript-eslint/prefer-namespace-keyword)
ES2015 module syntax is preferred over custom TypeScript modules and namespaces.ESLint(@typescript-eslint/no-namespace)

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `13.2.0` |
| `@typescript-eslint/parser`            | `5.17.0` |
| `TypeScript`                           | `4.5.2` |
| `ESLint`                               | `8.16.0` |
| `node`                                 | `14.18.2` |

**Comment by Josh-Cena** [comment]
"ES2015 module syntax" means

```ts
export const CompanyContract = 'test';
```

Without either `module` or `namespace`.

So [`no-namespace`](https://typescript-eslint.io/rules/no-namespace) says you should use neither `namespace` nor TS `module` keyword. [`prefer-namespace-keyword`](https://typescript-eslint.io/rules/prefer-namespace-keyword) says if you do have to use in-file modules, prefer the `namespace` keyword. You can check out the [TS docs](https://www.typescriptlang.org/docs/handbook/namespaces.html):

> Additionally, anywhere the `module` keyword was used when declaring an internal module, the `namespace` keyword can and should be used instead. This avoids confusing new users by overloading them with similarly named terms.

**Comment by Lonli-Lokli** [comment]
Got you, thanks,
